src/Microsoft.DotNet.Web.ProjectTemplates/content/StarterWeb-CSharp/Views/Shared/_CookieConsentPartial.cshtml

```diff
@@ -0,0 +1,65 @@
+﻿@using Microsoft.Extensions.Options
+@using Microsoft.AspNetCore.Builder
+@using Microsoft.AspNetCore.Http.Features
+@using Microsoft.Net.Http.Headers
+
+@inject IOptions<CookiePolicyOptions> CookiePolicy
+
+@{
+    var showBanner = (!Context.Features.Get<ITrackingConsentFeature>()?.CanTrack ?? false) && CookiePolicy.Value.ConsentCookie != null;
+
+    // Might be nice if the feature/builder could simply generate and return the cookie string for use in JS for me
+
+    var cookieName = CookiePolicy.Value.ConsentCookie?.Name;
+    var cookieValue = "yes"; // NOTE: Replace this with value from incoming public constant, see https://github.com/aspnet/Security/issues/1590
+    var options = CookiePolicy.Value.ConsentCookie?.Build(Context);
+
+    var setCookieHeaderValue = new SetCookieHeaderValue(
+        Uri.EscapeDataString(cookieName),
+        Uri.EscapeDataString(cookieValue))
+    {
+        Domain = options.Domain,
+        Path = options.Path,
+        Expires = options.Expires,
+        MaxAge = options.MaxAge,
+        Secure = options.Secure,
+        SameSite = (Microsoft.Net.Http.Headers.SameSiteMode)options.SameSite,
+        HttpOnly = options.HttpOnly
+    };
+
+    var cookieString = setCookieHeaderValue.ToString();
+}
```

Comment:
All the stuff above should not be in the view. I would suggest all this logic is encapsulated in a service somewhere.

Reply:
All of the things you commented on are going away as soon as I can consume the new feature from the linked issue. Universe is still building.